lhotse/recipes/csj.py

```diff
@@ -0,0 +1,246 @@
+import logging
+from concurrent.futures.thread import ThreadPoolExecutor
+from pathlib import Path
+from typing import Dict, List, Sequence, Tuple, Union
+
+from tqdm.auto import tqdm
+
+from lhotse import validate_recordings_and_supervisions, CutSet
+from lhotse.audio import Recording, RecordingSet
+from lhotse.recipes.utils import manifests_exist, read_manifests_if_cached
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike
+
+import random
+
+"""
+Corpus owner: https://clrd.ninjal.ac.jp/csj/en/index.html
+Corpus description: 
+- http://www.lrec-conf.org/proceedings/lrec2000/pdf/262.pdf
+- https://isca-speech.org/archive_open/archive_papers/sspr2003/sspr_mmo2.pdf 
+
+This script assumes that the transcript directory that was passed in has been parsed by csj_make_transcript.py. 
+Individual speaker IDs - or more precisely, session IDs, to cater for the 'D' dialogue cases - each have their own 
+folder. These are omitted as '...' in the directory tree below. 
+Notice that the 'D' transcripts are split into respective (L)eft and (R)ight channels.
+
+{transcript_dir}
+ - excluded
+   - ...
+ - core 
+   - ...
+ - eval1
+   - ...
+ - eval2
+   - ...
+ - eval3 
+   - ...
+ - noncore 
+   - ...
+   - A01F0576
+     - A01F0576.sdb (not used in this script)
+     - A01F0576-{transcript_mode}.txt
+     - A01F0576-segments (not used in this script)
+     - A01F0576-wav.list 
+   - ...
+   - D03M0038
+     - D03M0038.sdb (not used in this script)
+     - D03M0038-L-{transcript_mode}.txt
+     - D03M0038-L-segments (not used in this script)
+     - D03M0038-L-wav.list
+     - D03M0038-R-{transcript_mode}.txt
+     - D03M0038-R-segments (not used in this script)
+     - D03M0038-R-wav.list
+
+This script follows the espnet method of splitting the remaining core+noncore utterances into valid and train cutsets at index 4000.
+
+In other words, after moving the eval1, eval2, eval3, and excluded utterances out, the remaining utterances within the core and noncore 
+utterances are shuffled. The first 4000 utterances of the shuffled set go to the `valid` cutset and are not subjected to speed 
+perturbation. The remaining utterances become the `train` cutset and are speed-perturbed (0.9x, 1.0x, 1.1x) before saved to disk.  
+
+These cutsets serve as blueprints only. Fbank extraction is left for compute_fbank_csj.py.
+
+"""
+
+ORI_DATA_PARTS = (
+    "eval1",
+    "eval2",
+    "eval3",
+    "core",
+    "noncore",
+)
+
+SPLIT_DATA_PARTS = (
+    "train",
+    "valid",
+    "eval1",
+    "eval2",
+    "eval3"
+)
+
+RNG_SEED = 42
+
+def parse_transcript_header(line : str):
+    sgid, start, end, line = line.split(' ', maxsplit=3)
+    return (sgid, float(start), float(end), line)
+
+def parse_one_recording(
+    template : str,
+    wavlist_path : Path, 
+    recording_id : str
+) -> Tuple[Recording, List[SupervisionSegment]]:
+    transcripts = []
+    
+    for trans in glob(template + '*.txt'):
```

Comment:
I recommend using `pathlib.Path` for path operations (then you can write sth like `template.glob("*.txt")`)

Reply:
Because `template` is an incomplete path, I can't make `Path.glob` find the files appended with transcript type. 
Another reason is that I extract the transcript type from the resulting filename string. From a `Path` object, I will need to first convert it the path to a string. 

Though admittedly, I think globbing like this might break on Windows. 

```python
> from pathlib import Path
> partial_filename = Path("/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051/A04M0051")
> list(partial_filename.glob("*.txt"))
[]
> intended_result = Path("/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051")
> list(intended_result.glob("*.txt"))
[PosixPath('/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051/A04M0051-fluent.txt'), PosixPath('/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051/A04M0051-disfluent.txt'), PosixPath('/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051/A04M0051-number.txt'), PosixPath('/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051/A04M0051-symbol.txt')]
```

Reply:
How about the following:
```
dir = Path("/mnt/minami_data_server/t2131178/corpus/CSJ/retranscript_new/eval1/A04M0051")
partial_filename = dir.stem
list(dir.glob(f"{partial_filename}*.txt"))
```

Reply:
Thanks! It slipped my mind that I already have the spk_id. 
I've removed the reliance on `glob.glob`.